osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchHitExplosion.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Legacy
+{
+    public class LegacyCatchHitExplosion : CatchHitExplosion
+    {
+        private readonly Sprite explosion1;
+        private readonly Sprite explosion2;
+
+        [Resolved]
+        private Bindable<Color4> objectColour { get; set; }
+
+        [Resolved(Name = "CatcherWidth")]
+        private Bindable<float> catcherWidth { get; set; }
+
+        [Resolved]
+        private Bindable<float> catchPosition { get; set; }
```

Comment:
We generally don't do this. You'd be better off caching the whole drawable (catcher) or a model class which contains these bindables failing that.